src/test/java/com/thebois/models/world/MockWorld.java

```diff
@@ -0,0 +1,40 @@
+package com.thebois.models.world;
+
+import com.thebois.models.world.resources.IResource;
+import com.thebois.models.world.terrains.Grass;
+import com.thebois.models.world.terrains.ITerrain;
+
+public class MockWorld extends World {
```

Comment:
style

Should probably be called `TestWorld` and not `MockWorld` since it is not actually a mocked world. It's a real world, modified so it can be used for testing. A mock is usually a fake.